appengine/standard_python3/django/mysite/settings.py

```diff
@@ -59,10 +60,22 @@
 # Change this to "False" when you are ready for production
 DEBUG = env("DEBUG")
 
-# SECURITY WARNING: App Engine's security features ensure that it is safe to
-# have ALLOWED_HOSTS = ['*'] when the app is deployed. If you deploy a Django
-# app not on App Engine, make sure to set an appropriate host here.
-ALLOWED_HOSTS = ["*"]
+
+# [START gaestd_py_django_csrf]
+# SECURITY WARNING: It's recommended that you use this when
+# running in production. The URL will be known once you first deploy
+# to App Engine. This code takes the URL and converts it to both these settings formats.
+APPENGINE_URL = env("APPENGINE_URL", default=None)
+if APPENGINE_URL:
+    # Ensure the HTTPS is in the URL before it's used.
+    APPENGINE_URL = urlparse(APPENGINE_URL, "https").geturl()
+
+    ALLOWED_HOSTS = [APPENGINE_URL]
+    CSRF_TRUSTED_ORIGINS = [urlparse(APPENGINE_URL).netloc]
```

Comment:
@glasnt - could this line could simply be?
`CSRF_TRUSTED_ORIGINS = [APPENGINE_URL]`
 as per Django 4.0 updates to this (CSRF_TRUSTED_ORIGINS) setting (vs. [The values in older versions must only include the hostname (possibly with a leading dot) and not the scheme or an asterisk.](https://docs.djangoproject.com/en/4.0/ref/settings/#csrf-trusted-origins:~:text=The%20values%20in%20older%20versions%20must%20only%20include%20the%20hostname%20,%20and%20not%20the%20scheme%20or%20an%20asterisk)

Reply:
You're right here, I got the order wrong in this sample for which value needed the scheme and which didn't. Good catch!